### PR TITLE
[expat] Add ushort as char_type

### LINKED ISSUE #5975: expat missing option for char_type
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **expat/2.4.1**
  * Operating System+version: **Linux**
  * Compiler+version: **GCC 8**
  * Conan version: **conan 1.18.0**

### Description
Currently expat recipe accepts the following options
options = {"shared": [True, False], "fPIC": [True, False], "char_type": ["char", "wchar_t"]}

for char_type possible value should be 
"char_type": ["char", "wchar_t", "ushort"]

to be aligned with expat feature:
// Character type to use (char|ushort|wchar_t) [default=char]
EXPAT_CHAR_TYPE:STRING=char

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **expat/2.4.1**

fixes #5975

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/expat/all/conanfile.py [hunk 1]
```diff
@@ -79,5 +79,7 @@ def package_info(self):
         self.cpp_info.libs = tools.collect_libs(self)
         if not self.options.shared:
             self.cpp_info.defines = ["XML_STATIC"]
+        if self.options.get_safe("char_type") in ["wchar_t", "ushort"]:
+            self.cpp_info.defines.append("XML_UNICODE")
```
Comment (uilianries) [inline recipes/expat/all/conanfile.py:83]: https://github.com/libexpat/libexpat/blob/R_2_4_1/expat/CMakeLists.txt#L221

File: recipes/expat/all/conanfile.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ class ExpatConan(ConanFile):
     homepage = "https://github.com/libexpat/libexpat"
     license = "MIT"
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False], "char_type": ["char", "wchar_t"]}
+    options = {"shared": [True, False], "fPIC": [True, False], "char_type": ["char", "wchar_t", "ushort"]}
```
Comment (uilianries) [inline recipes/expat/all/conanfile.py:14]: https://github.com/libexpat/libexpat/blob/R_2_4_1/expat/CMakeLists.txt#L140

Comment (SpaceIm) [reply recipes/expat/all/conanfile.py:14]: maybe split this in one line per option, it becomes hard to read ;)

Comment (uilianries) [reply recipes/expat/all/conanfile.py:14]: It proves that you are human!

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**